### PR TITLE
Support ragged arrays

### LINKED ISSUE #108: Ragged arrays
It is currently assumed that all data sections are presented as tabular data, which is then conveniently parsed using `np.genfromtxt`.

This may be too restrictive for more general VRP variants. For instance, the site-dependent VRP restricts some vehicles from visiting a subset of clients. If we keep the assumption that data sections are tabular, then this would require a $N$ times $K$ matrix with 0 and 1's to denote whether client $i$ can be visited by vehicle $j$ or not. 

A more compact option would allow a list of indices. For example, `1 1 2 3 4` could mean that vehicle 1 is not allowed to visit clients 1 2 3 4.

### PR DESCRIPTION
**Pull request by leonlan**
This PR closes #108.

### CODE REVIEW

File: tests/parse/test_parse_vrplib.py [hunk 1]
```diff
@@ -126,6 +126,10 @@ def test_parse_specification(line, key, value):
             ["UNKNOWN_SECTION", "1 1", "1 -1"],
             ["unknown", np.array([1, -1])],
         ),
+        (
+            ["VEHICLES_ALLOWED_CLIENTS_SECTION", "1 2 3 4", "2 4 5", "3 6"],
+            ["vehicles_allowed_clients", [[2, 3, 4], [4, 5], [6]]],
```
Comment (N-Wouda) [inline tests/parse/test_parse_vrplib.py:131]: Vehicle-depot assignments use the `VEHICLES_DEPOT_SECTION` section name. Perhaps `VEHICLES_CLIENTS_SECTION`, in line with vehicle-depot assignments?

Comment (N-Wouda) [reply tests/parse/test_parse_vrplib.py:131]: Nvm this should be fine. Vehicles + clients is more confusing.

**Comment by leonlan** [comment]
This is just a quick patch to support ragged arrays because it's handy for testing PyVRP with site-dependent VRP instances. After this, I think we should only add support #117 before starting the big rehaul #114.

**Review by N-Wouda — APPROVED** [review]
LGTM

**Pull request merged**